src/utils/privileges.cpp

```diff
@@ -249,10 +250,14 @@ void verify_privileges_are_as_for_mainloop() {
 }
 
 
-void drop_root() {
+std::string drop_root(bool home_always_env) {
+	std::string home_dir;
 	#ifdef ANTINET_linux
 	try {
-		security_drop_root_from_sudo();
+		if (security_drop_root_from_sudo() && home_always_env) {
+			struct passwd *pw = getpwuid(getuid());
```

Comment:
hm meaning of this var is reversed here?

var is named "home_always_env" but actually seem to mean "home_ignore_env" because it instead reads user's passwd home